### PR TITLE
Added 'done' parameter to use default arguments

### PR DESCRIPTION
**Pull request by TechieBoy**
For #11

### CODE REVIEW

**Comment by praveensvsrk** [comment]
Why are you doing this?
```python
 if done_keyword in kwargs_store:
     done_evaluation = kwargs_store.pop(done_keyword,False)
```
What if the 'done' keyword is an **argument** of the function being curried?

**Comment by TechieBoy** [comment]
We can change the argument name to something like "curry_done=True" to avoid such conflicts unless there is some way to distinguish whether the argument is for the original function or for the curried function.

**Comment by akspi** [comment]
I think you also need to check for the case where you pass all the required arguments in one go along with the done=True condition. For example,
```python
def test_default_param2(self):
        @curry
        def add(a,b=10): 
            return a+b
        self.assertEqual(20,add(10,done=True))
```
Python should be throwing a TypeError for this case ( correct me if I'm wrong ). 
A possible fix for this is to add 
```python
if done_eval():
    return fun(*args_store, **kwargs_store)
```
immediately after your done_eval() definition.

File: curry.py [hunk 1]
```diff
@@ -28,18 +28,32 @@ def curried_factory(*initial_args, **initial_kwargs):
...
+            """
+            If done_keyword is present in kwargs_store, removes it 
+            from kwargs_store and returns it otherwise returns False.
+            """
+            done_keyword = "done"
+            done_evaluation = False
+            if done_keyword in kwargs_store:
+                done_evaluation = kwargs_store.pop(done_keyword,False)
+            return done_evaluation
+
```
Comment (akspi) [inline curry.py]: You'll need this line of code after your function definition.
```python
if done_eval():
    return fun(*args_store, **kwargs_store)
```

**Review by akspi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**